docs/gaia/gaia.rst

```diff
@@ -0,0 +1,337 @@
+
+::
+
+  from astroquery.gaia import Gaia
+  
+  upload_resource = 'my_table.xml'
+  j = Gaia.launch_job(query="select * from tap_upload.table_test", upload_resource=upload_resource, \
+  upload_table_name="table_test", verbose=True)
+  r = j.get_results()
+  print (r)
+
+
+1.6 Asynchronous query
+~~~~~~~~~~~~~~~~~~~~~~
+
+Asynchronous queries save results at server side. These queries can be accessed at any time. For anonymous users, results are kept for three days.
+
+The results can be saved in memory (default) or in a file.
+
+Query without saving results in a file:
+
+::
+
+  from astroquery.gaia import Gaia
+
+  job = Gaia.launch_job_async("select top 100 * from gaiadr1.gaia_source order by source_id")
+  
+  print (job)
+  r = job.get_results()
+  print (r['solution_id'])
+
+Query saving results in a file:
+
+::
+
+  from astroquery.gaia import Gaia
+  
+  job = Gaia.launch_job_async("select top 100 * from gaiadr1.gaia_source order by source_id", dump_to_file=True)
+  
+  print (job)
+  r = job.get_results()
+  print (r['solution_id'])
+
+
+1.6 Asynchronous job removal
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To remove asynchronous
+
+::
+
+  from astroquery.gaia import Gaia
+  
+  job = Gaia.remove_jobs(["job_id_1","job_id_2",...])
+
+
+---------------------------
+2. Authenticated access
+---------------------------
+
+Authenticated users are able to access to TAP+ capabilities (shared tables, persistent jobs, etc.)
+In order to authenticate a user, ``login`` or ``login_gui`` methods must be called. After a successful
+authentication, the user will be authenticated until ``logout`` method is called.
+
+All previous methods (``query_object``, ``cone_search``, ``load_table``, ``load_tables``, ``launch_job``) explained for
+non authenticated users are applicable for authenticated ones.
+
+The main differences are:
+
+* Asynchronous results are kept at server side for ever (until the user decides to remove one of them).
+* Users can access to shared tables.
+
+
+2.1. Login/Logout
+~~~~~~~~~~~~~~~~~
+
+Graphic interface
+
+
+*Note: Tkinter module is required to use login_gui method.*
+
+::
+
+  from astroquery.gaia import Gaia
+  
+  Gaia.login_gui()
+
+
+Command line
+
+
+::
+
+  from astroquery.gaia import Gaia
+  
+  Gaia.login(user='userName', password='userPassword')
+
+
+It is possible to use a file where the credentials are stored:
+
+*The file must containing user and password in two different lines.*
+
+::
+
+  from astroquery.gaia import Gaia
+  
+  Gaia.login(credentials_file='my_credentials_file')
+
+
+
+To perform a logout
+
+
+::
+
+  from astroquery.gaia import Gaia
+  
+  Gaia.logout()
+
+
+
+2.2. Listing shared tables
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+  from astroquery.gaia import Gaia
+
+  tables = Gaia.load_tables(only_names=True, include_shared_tables=True)
+  for table in (tables):
+    print (table.get_qualified_name())
+  
+
+-------------------------------------------
+3. Using TAP+ to connect other TAP services
```

Comment:
The fact that this is possible implies that tap should be its own module under `astroquery/utils` or maybe just under `astroquery`.  The documentation also deserves its own page.

Could you move all the tap modules to their own directory?